### PR TITLE
Adds clang-format to jenkinsfile as a step

### PR DESCRIPTION
**Pull request by SteveBronder**
#### Submisison Checklist

- [ ] Run tests: `./runCmdStanTests.py src/test`
- [ ] Declare copyright holder and open-source license: see below

#### Summary:

Adds the clang-format jenkins utility like `stan` and `math` have.

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company): Steve Bronder

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

**Comment by SteveBronder** [comment]
Are people okay with this? I was messing with this today and thought it would be nice to have since we have it in `stan` and `math`

**Comment by SteveBronder** [comment]
@serban-nicusor-toptal it looks like the windows machine can't find make?

https://jenkins.mc-stan.org/blue/organizations/jenkins/CmdStan/detail/PR-866/6/pipeline

```
C:\Jenkins2\workspace\CmdStan_PR-866>mingw32-make -j16 build 
```

**Comment by rok-cesnovar** [comment]
Needs a conflict resolve then good to go.

@serban-nicusor-toptal just so you are aware:
```
C:\Jenkins2\workspace\CmdStan_PR-866>mingw32-make -j16 build 
'mingw32-make' is not recognized as an internal or external command,
operable program or batch file.
script returned exit code 1
```

File: src/test/interface/metric_test.cpp [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+#include <stan/callbacks/stream_writer.hpp>
+#include <stan/services/error_codes.hpp>
+#include <stdexcept>
 #include <string>
-#include <test/utility.hpp>
 #include <test/test-models/proper.hpp>
-#include <stdexcept>
-#include <boost/math/policies/error_handling.hpp>
-#include <stan/callbacks/stream_writer.hpp>
+#include <test/utility.hpp>
```
Comment (rok-cesnovar) [inline src/test/interface/metric_test.cpp:9]: Do we have this header order stuff in Math also?

I personally more prefer the order we have in math:
- stan include
- std:: includes alphabetically.

Comment (rok-cesnovar) [reply src/test/interface/metric_test.cpp:9]: Everything else apart from this looks great!

**Comment by serban-nicusor-toptal** [comment]
Good morning, no worries!
Should be fixed now, there were some network issue including github too. Checked the machine and is fine too.

File: src/test/interface/metric_test.cpp (hunk 1 above)
Comment (SteveBronder) [reply src/test/interface/metric_test.cpp:9]: Oh weird I thought this would be respected via the .clang-format. Let me look at this

**Pull request merged**